hagglehaul.Server/Controllers/RiderController.cs

```diff
@@ -319,5 +319,36 @@ public async Task<IActionResult> ConfirmDriver([FromBody] AddTripDriver addTripD
             await _tripService.UpdateAsync(addTripDriver.TripId, trip);
             return Ok();
         }
+
+        [Authorize]
+        [HttpPost]
+        [Route("rating")]
+        public async Task<IActionResult> RateDriver([FromBody] GiveRating giveRating)
+        {
+            ClaimsPrincipal currentUser = this.User;
+
+            if (currentUser == null) { return BadRequest(new { Error = "Invalid User/Auth" }); };
+
+            var role = currentUser.FindFirstValue(ClaimTypes.Role);
+
+            if (role != "rider") { return Unauthorized(); }
+
+            DriverProfile driver = await _driverProfileService.GetAsync(giveRating.TargetUserEmail);
+            if (driver == null) { return BadRequest(new { Error = "Driver does not exist" }); }
+
+            if (driver.Rating == null)
+                driver.Rating = 0;
+            if (driver.NumRatings == null)
+                driver.NumRatings = 0;
+
+            var totalRatings = driver.Rating * (double)driver.NumRatings;
+
+            driver.NumRatings++;
+
+            driver.Rating = (totalRatings + (double)giveRating.RatingGiven) / (double)driver.NumRatings;
+
+            await _driverProfileService.UpdateAsync(driver.Email, driver);
```

Comment:
per the above comment, update the `Trip` object here as well (of course, checking that the trip time is 1. in the past 2. has a driver 3. the rider is the rider of the trip)